services/jitpack/jitpack-downloads.service.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+const { downloadCount } = require('../color-formatters')
+const { metric } = require('../text-formatters')
+
+const schema = Joi.object({
+  week: nonNegativeInteger,
+  month: nonNegativeInteger,
+}).required()
+
+const periodMap = {
+  dw: {
+    api_field: 'week',
+    suffix: '/week',
+  },
+  dm: {
+    api_field: 'month',
+    suffix: '/month',
+  },
+}
+
+module.exports = class JitpackDownloads extends BaseJsonService {
+  static get examples() {
+    return [
+      {
+        title: 'JitPack - Downloads',
+        namedParams: {
+          groupId: 'jitpack',
+          artifactId: 'maven-simple',
+          period: 'dm',
+        },
+        staticPreview: JitpackDownloads.render({
+          downloads: 14000,
+          period: 'dm',
+        }),
+        keywords: ['java', 'maven'],
+      },
+    ]
+  }
+
+  static get category() {
+    return 'downloads'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'downloads' }
+  }
+
+  static get route() {
+    return {
```

Comment:
I couldn't get the inline suggestion to work for multiple lines, but in order to be consistent with other download badges we should drop the `/d` from the base and move `period` up as the first parameter, i.e.
```js
    return {
      base: 'jitpack',
      pattern: ':period(dw|dm)/:groupId/:artifactId',
    }
```

Reply:
Strangely, when I set `jitpack/v` to `jitpack` only, it apparently fails all tests (including the former Jitpack version implementation) probably because of [this line](https://github.com/erayerdin/shields/blob/master/services/jitpack/jitpack.service.js#L21).

That is why I've set it as `jitpack/d`, to be more consistent with the former version implementation.

Let me have a further look into this. Maybe I'm missing something?

Reply:
All tests fail due to the errors similar to below:

```
AssertionError: expected { Object (name, value) } to deeply equal { Object (name, value) }                                                          
      + expected - actual

       {
         "name": "jitpack"
      -  "value": "inaccessible"
      +  "value": "project not found or private"
       }
```

Should I commit the failing codebase for you to have a look?

Reply:
Okay, okay, I think I've done it. It was my stupidity. I was just passing `period` to `render` method without its name. I'll push the new commits regarding to the changes that you've requested.